back-end/server.js

```diff
@@ -72,11 +73,37 @@ class Task {
     constructor(taskId, taskInfo, contract, images) {
         this.taskId = taskId;
         this.taskInfo = taskInfo;
-        this.contract = contract;
+        // this.contract = contract;
+        this.taskContract = new ethers.Contract(
+            contract,
+            TaskContract.abi,
+            wallet
+        );
+        this.settlementContract = new ethers.Contract(
+            this.taskContract.settlement(),
+            SettlementContract.abi,
+            wallet,
+        );
+
+        // Register event listeners
+        const settleEvent = {
+            address: this.taskContract.address,
+            topics: [
+                ethers.utils.id('Settle(uint)'),
+            ]
+        };
+        const disperseEvent = {
+            address: this.settlementContract.address,
+            topics: [
+                ethers.utils.id('Deposit(address,uint)'),
+            ]
+        };
+        this.settlementContract.on(settleEvent, this.settleContract.bind(this));
+        this.settlementContract.on(disperseEvent, () => { console.log('disperse success!') });
```

Comment:
we need a 'new funds deposited' event to update the UI (or to call contract.getBalance() periodically).